benchmark/datasets.py

```diff
@@ -93,8 +93,17 @@ def ivecs_read(fname):
 
 def xbin_mmap(fname, dtype, maxn=-1):
     """ mmap the competition file format for a given type of items """
-    n, d = map(int, np.fromfile(fname, dtype="uint32", count=2))
-    assert os.stat(fname).st_size == 8 + n * d * np.dtype(dtype).itemsize
+    # for search
+    # n, d = map(int, np.fromfile(fname, dtype="uint32", count=2))
+    # for dim reduced file of dtype float32
+    n, d = map(int, np.fromfile(fname, dtype="float32", count=2))
+    n = 100000000
+    d = 32
```

Comment:
This shouldn't be here. (And it seems the whole `xbin_mmap` should not be changed at all.

Reply:
yes, absolutely -- these had to be hardcoded on my side, because the derived values were not correct. I will remove this.